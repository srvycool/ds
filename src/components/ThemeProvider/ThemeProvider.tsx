import React from 'react';
import { ThemeProvider as SCThemeProvider } from 'styled-components';
import { theme } from '../../theme/theme';

export const ThemeProvider: React.FC = ({ children }) => {
  return <SCThemeProvider theme={theme}>{children}</SCThemeProvider>;
};
