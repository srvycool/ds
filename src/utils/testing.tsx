import React from 'react';
import {
  render as tlrender,
  RenderOptions,
  RenderResult,
} from '@testing-library/react';
import { ThemeProvider } from '../components/ThemeProvider/ThemeProvider';

export function render(
  ui: React.ReactElement,
  options?: Omit<RenderOptions, 'queries'>
): RenderResult {
  return tlrender(<ThemeProvider>{ui}</ThemeProvider>, options);
}
