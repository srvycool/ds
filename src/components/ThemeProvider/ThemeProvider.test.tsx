import React from 'react';
import { render } from '../../utils/testing';
import { ThemeProvider } from './ThemeProvider';

describe('ThemeProvider', () => {
  it('should match snapshot', () => {
    const { container } = render(
      <ThemeProvider>I am a children</ThemeProvider>
    );
    expect(container).toMatchSnapshot();
  });
});
