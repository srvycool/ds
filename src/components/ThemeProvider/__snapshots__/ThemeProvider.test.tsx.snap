// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThemeProvider should match snapshot 1`] = `
<div>
  I am a children
</div>
`;
